### PR TITLE
284 clear proofs

### PR DESCRIPTION
**Pull request by AnushaTiwari5**
Clear Proof and Proof Drag working

### CODE REVIEW

**Comment by AnushaTiwari5** [comment]
Looks like it might solve issue in pr #279

File: src/ProofTools/DeiterationTool.ts [hunk 1]
```diff
@@ -30,7 +30,10 @@ let currentProofTree: AEGTree;
  */
 export function deiterationMouseDown(event: MouseEvent) {
     const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
-    currentProofTree = new AEGTree(treeContext.getLastProofStep().tree.sheet);
+    currentProofTree = new AEGTree();
+    if (treeContext.currentProofStep) {
+        currentProofTree.sheet = treeContext.currentProofStep.tree.sheet.copy();
+    }
```
Comment (James-Oswald) [inline src/ProofTools/DeiterationTool.ts:36]: If we need this exact block of code in every single proof mode tool it should be turned into a helper function.

**Review by James-Oswald — APPROVED** [review]

**Comment by DawnTheWitch** [comment]
Because this branch was the merged version of drag in proof and proof history I just added my fix that was needed on PR #264. This is essentially the PRs #264, #279 along with the issue for clear canvas.

File: src/ProofHistory.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        case "Iteration":
+            icon.className = "fa fa-expand";
+            break;
+        case "Deiteration":
+            icon.className = "fa fa-compress";
+            break;
+        case "Pasted":
+            icon.className = "fa fa-files-o";
+            break;
+    }
```
Comment (James-Oswald) [inline src/ProofHistory.ts:63]: Using an inline dictionary here would make this clearer. IE
``` 
icon.className = "fa fa-" + {
"Single Move" :  "mouse-pointer",
"Multi Move" : "arrows"
...
}[newStep.appliedRule];

File: src/ProofTools/DeiterationTool.ts (hunk 1 above)
Comment (James-Oswald) [reply src/ProofTools/DeiterationTool.ts:36]: Still holds

File: src/index.css
Comment (James-Oswald) [inline src/index.css:1]: Fix indentation. CSS for this project should use 4 spaces for indentation. I see this was pushed on Subrina's branch and I didn't catch this, please anything in this file without 4 spaces.

**Review by James-Oswald — DISMISSED** [review]

File: src/ProofTools/DeiterationTool.ts (hunk 1 above)
Comment (DawnTheWitch) [reply src/ProofTools/DeiterationTool.ts:36]: This has been added to the master clean up issue.

File: src/ProofHistory.ts [hunk 3]
```diff
@@ -29,38 +29,20 @@ export function appendStep(newStep: ProofNode) {
...
-        case "Pasted":
-            icon.className = "fa fa-files-o";
-            break;
-    }
+    icon.className =
+        "fa fa-" +
+        {
+            "Single Move": "mouse-pointer",
+            "Multi Move": "arrows",
+            Resize: "arrows-alt",
```
Comment (James-Oswald) [inline src/ProofHistory.ts:37]: I'm suspicious that this may not work without quotations, though its possible it will, add them anyway for quality.

**Review by James-Oswald — APPROVED** [review]

**Pull request merged**